### DISCUSSION TITLE
How to fix "Store is not injected"

### DISCUSSION THREAD

**Discussion by SalzstangeSalamiBrief** [quasarframework/quasar#11091] [General - Components / Directives / etc] [answered]
Hello,
I created a new Quasar App with Vuex enabled and tried to access the store inside the setup-Method of the Index.vue Page as described in the documentation. But i am not able to access the Store. I get the warning "injection "store" not found  at <PageIndex>...." and if i debug the File, then the store is undefined.

![image](https://user-images.githubusercontent.com/33378894/138674935-acf4f440-fa8f-451b-933b-7912bdf2bfde.png)

First I tried to update my dependencies. After that i added the store to Vue inside the autogenerated file ./quasar/app.js. With that i am able to access the store but i receive the warning, that the store is already registered.
Additionally i tried to create a new Quasar App with the kit vue-ts and Vuex enabled. In This project i added the example store-module to the store and tried to access the store inside my Component. The Result is the same as before. The relevant/modified files are these:

./store/index.ts
![image](https://user-images.githubusercontent.com/33378894/138675427-fa471844-b318-4a85-993f-bbde60d2dd24.png)

./pages/Index.vue
![image](https://user-images.githubusercontent.com/33378894/138675554-47bd9dda-a15e-4b8d-8078-7c8bd16b01a8.png)

Dependencies:
![image](https://user-images.githubusercontent.com/33378894/138676044-62362888-384d-46bc-87e0-773541927e35.png)

Has someone an Idea how to fix this problem?

**Comment by MilosPaunovic** [comment]
Just to be sure, you've gonne through the entire VUEX section of the docs and could not find a solution?
https://quasar.dev/quasar-cli/vuex-store

One more thing to note, very important - you should never change anything in the `.qusaar` folder; it is auto-generated and every change in the code will rebuild the files and overwrite your changes.

**Comment by FelipeVeiga** [comment]
i solved that exporting the Store directly and then imporing Store
I think you can not use the funciotion store outside the setup.
i dont know if correct way but works.

**Comment by SalzstangeSalamiBrief** [comment, marked as answer]
Thank you both for the input.
I Read the Vuex docs as well the Quasar-Docs for Vuex and could not find any solution/cause of the problem.
To verify that i did not miss any steps i changed my component to use the Options-API.
Inside the Options-API i had no trouble accessing the store with this.$store.
After some trial and error and @FelipeVeiga hint i found out that the cause is the import-Statement:
in the docs  useStore gets imported from vuex. I changed the source of the import to the index.ts file of the Store and the problem disappeared. The import now looks like that:

![image](https://user-images.githubusercontent.com/33378894/138701777-30e7d251-1171-4847-b480-284db8049356.png)
